docs/src/design/proposals/APIRedesign.md

```diff
@@ -0,0 +1,1050 @@
+  * For example, call them "applications."
+* Consolidate what are currently called "representations" and
+  "templates", and instead have them be "views".
+
+For parity with actions, a `View` would be a constructor which
+takes an `ActionContext` as a parameter (with similarly-defined
+properties) and exposes a method to retrieve the HTML elements
+associateed with it.
+
+The platform would then additionally expose an `AngularView`
+implementation to improve compatibility with existing
+representations, whose usage would something like:
+
+```js
+define(
+  ["AngularView"],
+  function (AngularView) {
+    var template = "<span ng-click='...'>Hello world</span>";
+    return new AngularView(template);
+  }
+);
+```
+
+The interface exposed by a view is TBD, but should provide at
+least the following:
+
+* A way to get the HTML elements that are exposed by & managed
+  by the view.
+* A `destroy` method to detach any listeners at the model level.
+
+Individual views are responsible for managing their resources,
+e.g. listening to domain objects for mutation. To keep DRY, the
+platform should include one or more view implementations that
+can be used/subclassed which handle common behavior(s).
+
+### Benefits
+
+* Using Angular API for views is no longer required.
+* Views become less-coupled to domain objects. Domain objects
+  may be present in the `ViewContext`, but this also might
+  just be a "view" of some totally different thing.
+* Helps clarify high-level concerns in the API (a View is now
+  really more like a View in the MVC sense; although, not
+  completely, so this gets double-booked as a detriment.)
+* Having a `ViewContext` that gets passed in allows views to
+  be more "contextually aware," which is something that has
+  been flagged previously as a UX desire.
+
+### Detriments
+
+* Becomes less clear how views relate to domain objects.
+* Adds another interface.
+* Leaves an open problem of how to distinguish views that
+  a user can choose (Plot, Scrolling List) from views that
+  are used more internally by the application (tree view.)
+* Views are still not Views in the MVC sense (in practice,
+  the will likely be view-controller pairs.) We could call
+  them widgets to disambiguate this.
+* Related to the above, even if we called these "widgets"
+  it would still fail to enforce good MVC.
+
+## Wrap Angular Services
+
+Wrap Angular's services in a custom interfaces; e.g.
+replace `$http` with an `httpService` which exposes a useful
+subset of `$http`'s functionality.
+
+### Benefits
+
+* Removes a ubiquitous dependency on Angular.
+* Allows documentation for these features to be co-located
+  and consistent with other documentation.
+* Facilitates replacing these with non-Angular versions
+  in the future.
+
+### Detriments
+
+* Increases the number of interfaces in Open MCT Web. (Arguably,
+  not really, since the same interfaces would exist if exposed
+  by Angular.)
+
```

Comment:
I would add a detriment here that for established Angular developers this introduces a layer of obfuscation and uncertainty. If I use angular services like $http etc. I know they integrate with the digest cycle and I don't need to worry about managing that myself. If I saw an httpService, I wouldn't have the confidence that this was the case, and I would go digging around in the code to see how it's implemented.

Not saying that this approach is wrong, I totally understand the need to reduce dependencies on libraries, especially Angular. But it's worth bearing in mind.